packages/node/src/policy-gen/policy-gen-context.js

```diff
@@ -95,19 +88,35 @@ export class PolicyGeneratorContext {
    * A mapping of compartment names to missing module specifiers encountered
    * during the policy generation process.
    *
-   * @type {MissingModuleMap | undefined}
+   * @type {Readonly<Set<string>>}
    */
 
   #missingModules
 
-  /** @type {Loggerr} */
+  /** @type {Readonly<Loggerr>} */
   #log
 
   /**
    * Used for reading source files
    */
   static #decoder = new TextDecoder()
```

Comment:
What's the benefit of having it as a static field instead of a variable in higher scope?

Reply:
logical grouping and/or encapsulation.  if there was something in this module besides this class, it may not need this `TextDecoder` instance.  if it _did_ need a `TextDecoder` instance, then I would consider pulling it out into a higher scope.

if I'm working with classes, I will generally reach for a static field instead of a  scoped variable, for no other reason except it is easier for me to understand who owns what.